### PR TITLE
Assignment 1

### PR DESCRIPTION
**Pull request by pmkaboo**
I chose minitest for testing because its simple and fast, and we wont need any complex testing for any of these assignments.

### CODE REVIEW

**Comment by pmkaboo** [comment]
- Enhance the component with ability to print out the name and price of the product

does the print need some more formatting, or is `product.name/price` good enough?

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# A sample Gemfile
+source "https://rubygems.org"
+
+gem 'minitest'
```
Comment (martincik) [inline Gemfile]: Where did you forget the `Gemfile.lock`? Why didn't you commit it too?

File: product.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Product
+	attr_accessor :name, :price
+
+	def initialize name, price
+		@name = name
+		@price = price
```
Comment (martincik) [inline product.rb]: - What if the price is not a number?
- What type is the best to represent price/money?

File: product.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+	end
```
Comment (martincik) [inline product.rb]: I'm missing the ability to print name and price here. Can you add it?
You can print is as `#{name}, #{price}`. Price should be formatted as `XXXX.YY` where YY is only 2 after the dot as this is the standard for money.

File: Gemfile (hunk 1 above)
Comment (pmkaboo) [reply Gemfile]: i installed the gem manually and just later thought about creating gemfile, so i didnt run bundler and lock didnt get created

File: product.rb (hunk 2 above)
Comment (pmkaboo) [reply product.rb]: assignment 1 is just about printing, so it doesnt really matter if its a number. later on i changed it to float, which is not ideal for currency, but i think its enough for this example. in real world situation id use decimal.

Comment (martincik) [reply product.rb]: Ok. Having price in any code should imply that you would use decimal no matter if it's just printing example. Most importantly this is a simple test but should reflect real-world skills.

Can you rewrite this with decimal and also add the print as I've specified?

File: Gemfile (hunk 1 above)
Comment (martincik) [reply Gemfile]: Ok. So add it and make sure you'll not add half-done stuff in the repo next time. Just go through the code one-more-time before you commit. Making sure all is good for other developers to use it.

File: product.rb (hunk 2 above)
Comment (pmkaboo) [reply product.rb]: ok, you mentioned `XXXX.YY` format for the price, should we round the two decimals or just truncate them?

Comment (martincik) [reply product.rb]: What would you do if you would have an online e-commerce store? What would you like the customer to see? I will let you decide. I need you to show me your thinking.

**Comment by martincik** [comment]
RE: to_d and exception: Very good. What could you do better to prevent this misunderstanding from happening in the future?

**Comment by pmkaboo** [comment]
well, this doesnt seem like preventable issue, im using fresh installation of latest version of ruby(set it up on tuesday), so updating included gems isnt something i would think about.
maybe by including ruby version in gemfile i could lower the chance of this happening, but if either of us updated bigdecimal, it would still come up. 
overall, it just seems like an example of why code review is important.

**Pull request merged**